pkg/operator/operator.go

```diff
@@ -256,7 +259,56 @@ func New(logger logr.Logger, clientConfig *rest.Config, opts Options) (*Operator
 					},
 				}})
 		}),
-		CertDir: certDir,
+		TLSOpts: []func(*tls.Config){
+			func(c *tls.Config) {
+				var ca *x509.Certificate
+				rawCA, err := os.ReadFile(opts.ClientCAFile)
+				if err != nil {
+					if opts.CACert != "" {
+						var decErr error
+						rawCA, decErr = base64.StdEncoding.DecodeString(opts.CACert)
+						if decErr != nil {
+							logger.Error(decErr, "unable to decode CA Cert argument")
+							return
+						}
+					} else {
+						logger.Error(err, "unable to read client CA for webhook server, skipping")
+						return
+					}
+				}
+				ca, err = x509.ParseCertificate(rawCA)
+				if err != nil {
+					logger.Error(err, "unable to parse client CA for webhook server, skipping")
+					return
+				}
+				c.ClientCAs.AddCert(ca)
+			},
+			func(c *tls.Config) {
+				cert, err := tls.LoadX509KeyPair(opts.CertFile, opts.KeyFile)
+				if err != nil {
+					if opts.TLSCert != "" && opts.TLSKey != "" {
+						tlsCert, decErr := base64.StdEncoding.DecodeString(opts.TLSCert)
+						if decErr != nil {
+							logger.Error(decErr, "unable to decode TLS Cert")
+						}
+						tlsKey, decErr := base64.StdEncoding.DecodeString(opts.TLSKey)
+						if decErr != nil {
+							logger.Error(decErr, "unable to decode TLS Key")
+						}
+
+						var tlsErr error
+						cert, tlsErr = tls.X509KeyPair(tlsCert, tlsKey)
+						if tlsErr != nil {
+							logger.Error(tlsErr, "invalid cert or key provided")
+						}
+					} else {
+						logger.Error(err, "unable to load TLS certificate/key pair for webhook server, skipping")
+						return
+					}
+				}
+				c.Certificates = []tls.Certificate{cert}
```

Comment:
Should we maybe append here?

Reply:
Shouldn't matter in this case, but I'll make the change.